README.md

```diff
@@ -15,8 +15,80 @@ following entitlements are supported:
 
 ## Obtaining the Client
 
-The client comes pre-installed on all Ubuntu systems. Users can run the
-`ua` command to learn more or view the manpage.
+The client comes pre-installed on all Ubuntu systems in the debian packages
+`ubuntu-advantage-tools` and `ubuntu-advantage-pro`. Users can run the `ua` command to learn more or view
+the manpage.
+
+## Terminology
+ The following vocabulary is used to describe different aspects of the work Ubuntu Advantage Client performs:
+
+| Term | Meaning |
+| -------- | -------- |
+| UA Client | The python command line client represented in this ubuntu-advantage-client repository. It is installed on each Ubuntu machine and is the entry-point to enable any Ubuntu Advantage commercial service on an Ubuntu machine. |
+| Contract Server | The backend service exposing a REST API to which UA Client authenticates in order to obtain contract and commercial service information and manage which support servics are active on a machine.|
+| Entitlement/Service | An Ubuntu Advantage commercial support service such as FIPS, ESM, Livepatch, CIS-Audit to which a Contract may be entitled |
+| Affordance | Service-specific list of applicable architectures and Ubuntu series on which a service can run |
+| Directives | Service-specific configuration values which are applied to a service when enabling that service |
+| Obligations | Service-specific policies that must be instrumented for support of a service. Example: `enableByDefault: true` means that any attached machine **MUST** enable a service on attach |
+
+
+## Architecture
+Ubuntu Advantage client, hereafter "UA client", is python3-based command line utility. It provides a CLI to attach, detach, enable,
+disable and check status of support related services.
+
+The package `ubuntu-advantage-tools` also provides a C++ APT hook which helps advertise ESM service and available packages in MOTD and during various apt commands.
+
+The `ubuntu-advantage-pro` package delivers auto-attach auto-enable functionality via init scripts and systemd services for various cloud platforms.
+
+By default, Ubuntu machines are deployed in an unattached state. A machine can get manually or automatically attached to a specific contract by interacting with the Contract Server REST API to enable a machine. Any change in state of services or machinbe attach results in additional interactions with the Contract Server API to validate such operations. The contract server API is described by the [ua-contracts openapi spec](https://github.com/CanonicalLtd/ua-contracts/blob/develop/docs/contracts.yaml).
```

Comment:
Maybe we can change the sentence from: 

`A machine can get manually or automatically attached to a specific contract by interacting with the Contract Server REST API to enable a machine`

To:

`A machine can get manually or automatically attached to a specific contract by interacting with the Contract Server REST API`

Also, there is a typo on `machinbe`